lib/FrontendTool/LoadedModuleTrace.cpp

```diff
@@ -911,17 +911,34 @@ class ObjcMethodReferenceCollector: public SourceEntityWalker {
 };
 
 static std::optional<int> createObjCMessageTraceFile(const InputFile &input,
-                                                     ModuleDecl *MD) {
+                                                     ModuleDecl *MD,
+                                        std::vector<SourceFile*> &filesToWalk) {
+  if (input.getLoadedModuleTracePath().empty()) {
+    // we basically rely on the passing down of module trace file path
+    // as an indicator that this job needs to emit an ObjC message trace file.
+    // FIXME: add a separate swift-frontend flag for ObjC message trace path
+    // specifically.
+    return {};
+  }
+  for (auto *FU : MD->getFiles()) {
+    if (auto *SF = dyn_cast<SourceFile>(FU)) {
+      if (SF->getFilename().ends_with(".swift")) {
```

Comment:
IMO it would be better to check the `SourceFileKind`

Reply:
Thanks for taking a look! Address this in https://github.com/swiftlang/swift/pull/77396